### PR TITLE
Support for user lookup via NSS

### LINKED ISSUE #1402: Installation fails with nss-altfiles
### Before creating an issue, make sure you've checked the following:

- [X] You are running the latest released version of k0s
- [X] Make sure you've searched for existing issues, both open and closed
- [X] Make sure you've searched for PRs too, a fix might've been merged already
- [X] You're looking at docs for the released version, "main" branch docs are usually ahead of released versions.

### Version

v1.23.1+k0s.0

### Platform

```shell
Fedora CoreOS 32.20201104.3.0
```

### What happened?

As was foreshadowed in #1079
> forcing file based lookup will probably just break people's setups in a different way >:

the fix made there is breaking on rpm-ostree base distros (as far as I can tell that would be CoreOS/Flatcar/Silverblue) but the real issue is with [altfiles](https://github.com/aperezdc/nss-altfiles).

### Steps to reproduce

1. Install and configure nss-altfiles (or as in my case use Fedora Coreos)
2. Have one more of the installConfig.users defined in [/lib/passwd](https://github.com/coreos/fedora-coreos-config/blob/d1743185ffd47dce8fa5e885894d3a6a068b5dee/manifests/passwd#L16)
3. `k0s install controller`

### Expected behavior

exit 0, nor error

### Actual behavior

Errors with failure to create the user from step 2 (on a default fedora coreos this happens to etcd).
The user indeed exists so the error _to create_ is expected but this comes from the osusergo implementation not being able to find the user via nsss in the first place.

### Screenshots and logs

_No response_

### Additional context

The obvious workaround (once all the above troubleshooting was figured out) is to change the etcduser and create a new one
```
  installConfig:
    users:
      etcdUser: k0s-etcd
```
but it would be nice for it to work out of the box (other projects have resorted to having separate musl/glibc builds).

### PR DESCRIPTION
**Pull request by ncopa**
Add support to lookup users via NSS by calling external command  `id`.

Fixes #1402

### CODE REVIEW

File: internal/pkg/users/users.go [hunk 1]
```diff
@@ -16,26 +16,24 @@ limitations under the License.
...
-		return 0, err
+	if err == nil {
+		return strconv.Atoi(entry.Uid)
 	}
-	return strconv.Atoi(entry.Uid)
-}
-
-func CheckIfUserExists(name string) (bool, error) {
-	_, err := user.Lookup(name)
 	if _, ok := err.(user.UnknownUserError); ok {
```
Comment (s0j) [inline internal/pkg/users/users.go:31]: This can probably be `errors.Is(..)`

File: internal/pkg/users/users.go [hunk 2]
```diff
@@ -16,26 +16,24 @@ limitations under the License.
...
-		return false, nil
-	}
-	if err != nil {
-		return false, err
+		// fallback to call external `id` in case NSS is used
+		out, err := exec.Command("id", "-u", name).CombinedOutput()
```
Comment (s0j) [inline internal/pkg/users/users.go:33]: Can this be the full path to `id` instead?

Comment (ncopa) [reply internal/pkg/users/users.go]: It can.

File: internal/pkg/users/users.go (hunk 1 above)
Comment (ncopa) [reply internal/pkg/users/users.go]: good idea

File: pkg/install/users.go [hunk 3]
```diff
@@ -73,22 +73,11 @@ func DeleteControllerUsers(clusterConfig *v1beta1.ClusterConfig) error {
...
 func EnsureUser(name string, homeDir string) error {
-	exists, err := users.CheckIfUserExists(name)
-	if err != nil {
-		return err
-	}
-
-	if exists {
-		return nil
+	_, err := users.GetUID(name)
+	if _, ok := err.(user.UnknownUserError); ok {
```
Comment (s0j) [inline pkg/install/users.go:77]: I missed this originally -- this could be `errors.Is()` as well. Not holding this up though.

**Review by s0j — DISMISSED** [review]

File: pkg/install/users.go (hunk 3 above)
Comment (ncopa) [reply pkg/install/users.go:77]: doh... I think we should fix this too.

**Review by s0j — APPROVED** [review]

**Pull request merged**